Make ironic timeouts customizable

furkatgofurov7 opened the pull request:
**What this PR does / why we need it:**
This commit adds some parameters for the ironic configuration

furkatgofurov7 commented:
/cc @maelk

maelk commented:
I didn't see that it was set this way in ironic.conf, sorry, discard what I said, since we anyways set our own settings

furkatgofurov7 commented:
/test-integration

maelk commented on configure-ironic.sh:
```diff
@@ -13,6 +13,15 @@ IRONIC_FAST_TRACK=${IRONIC_FAST_TRACK:-true}
 IRONIC_AUTOMATED_CLEAN=${IRONIC_AUTOMATED_CLEAN:-true}
 
+# Timeout (seconds) to wait for a callback from a deploy ramdisk
+IRONIC_DEPLOY_CALLBACK_TIMEOUT=${IRONIC_DEPLOY_CALLBACK_TIMEOUT:-"4800"}
```
Considering the comment you pointed out, we might also want to put the pxe deploy timeout customized here with the same note about its dependency

furkatgofurov7 replied:
@maelk this is the comment I was referring to in the [ironic.conf file](https://github.com/metal3-io/ironic-image/blob/e1d39417015f5a7799a42a91a36f7005d872b562/ironic.conf#L55) and I have made it customizable as you suggested.

furkatgofurov7 commented:
/test-integration

maelk commented:
/lgtm

furkatgofurov7 commented:
/test-integration

furkatgofurov7 commented:
/assign @bfournie
/cc @elfosardo
/cc @hardys

elfosardo commented on configure-ironic.sh:
```diff
@@ -13,6 +13,20 @@ IRONIC_FAST_TRACK=${IRONIC_FAST_TRACK:-true}
 IRONIC_AUTOMATED_CLEAN=${IRONIC_AUTOMATED_CLEAN:-true}
 
+# Timeout (seconds) to wait for a callback from a deploy ramdisk
+IRONIC_DEPLOY_CALLBACK_TIMEOUT=${IRONIC_DEPLOY_CALLBACK_TIMEOUT:-"4800"}
```
the options look good, just the values seem a bit higher than expected, how did you get them?

furkatgofurov7 replied:
@elfosardo I was checking [ironic.conf file](https://github.com/metal3-io/ironic-image/blob/e1d39417015f5a7799a42a91a36f7005d872b562/ironic.conf#L31) and the values were set in that file. I guess it sets this value in purpose to make sure [boot_retry_timeout](https://github.com/metal3-io/ironic-image/blob/e1d39417015f5a7799a42a91a36f7005d872b562/ironic.conf#L56) is 3 times less than that variable as described in the comments. Maybe, we can set it to **"3600"** if it is higher than expected?

elfosardo replied:
I see, now I remember why we set the value like that. That's fine then, thanks!

furkatgofurov7 replied:
Thank you for reviewing.

furkatgofurov7 commented:
/cc @dhellmann
/cc @dtantsur

dhellmann commented:
I feel a bit like we're reimplementing confd. :-)

These are oslo.config settings, right? In other places we used the automatically checked environment variables to pass values right to the daemon. Would that approach work here, too? Could we document the relevant variables names instead of modifying the config file?

maelk commented:
That's a very good point. This change could instead be a readme explaining how to override the settings in the ironic config.

derekhiggins commented:
> That's a very good point. This change could instead be a readme explaining how to override the settings in the ironic config.

Something like this could work
https://github.com/metal3-io/ironic-image/pull/165

maelk commented on README.md:
```diff
@@ -35,3 +35,9 @@ The following environment variables can be passed in to customize run-time funct
 - DHCP_RANGE - dhcp range to use for provisioning (default 172.22.0.10-172.22.0.100)
 - MARIADB_PASSWORD - The database password
+
+The following environment variables can be used to override the settings in the ironic configuration file:
```
I would make it less specific I guess, saying that configuration can be overriden by environment variables. Then say as an example the following (and show the config options so that it is understandable how to extend this to other configuration options)

furkatgofurov7 replied:
Fixed, thank you for feedback

furkatgofurov7 commented:
> I feel a bit like we're reimplementing confd. :-)
> 
> These are oslo.config settings, right? In other places we used the automatically checked environment variables to pass values right to the daemon. Would that approach work here, too? Could we document the relevant variables names instead of modifying the config file?

@dhellmann thanks for the suggestion, I have addressed your comments and it is now only in the readme document.

furkatgofurov7 commented:
/test-integration

furkatgofurov7 commented:
/test-integration

derekhiggins commented:
> > I feel a bit like we're reimplementing confd. :-)
> > These are oslo.config settings, right? In other places we used the automatically checked environment variables to pass values right to the daemon. Would that approach work here, too? Could we document the relevant variables names instead of modifying the config file?
> 
> @dhellmann thanks for the suggestion, I have addressed your comments and it is documented now in the readme by following the format from #165

I think we should just merge #165, I don't see any reason why these Timeout values should be special and mentioned in the config.

dhellmann commented:
I don't have a problem with listing some variables that are commonly useful to override. Maybe we can also describe the general way to build the right name for other variables? That could be a separate PR.

This one needs a rebase.

furkatgofurov7 commented:
> > > I feel a bit like we're reimplementing confd. :-)
> > > These are oslo.config settings, right? In other places we used the automatically checked environment variables to pass values right to the daemon. Would that approach work here, too? Could we document the relevant variables names instead of modifying the config file?
> > 
> > 
> > @dhellmann thanks for the suggestion, I have addressed your comments and it is documented now in the readme by following the format from #165
> 
> I think we should just merge #165, I don't see any reason why these Timeout values should be special and mentioned in the config.



> I don't have a problem with listing some variables that are commonly useful to override. Maybe we can also describe the general way to build the right name for other variables? That could be a separate PR.
> 
> This one needs a rebase.

These are examples of timeout values and how can they be overridden. And also we are not mentioning them in the config but README.md file.

/test-integration

furkatgofurov7 commented:
Hi @dhellmann, is there anything that needs to be changed/improved in this PR, or we can move forward with it?

dhellmann commented:
/lgtm

furkatgofurov7 commented:
Hi @derekhiggins! is there anything that needs to be improved in this PR or we can proceed with it?

derekhiggins commented:
/lgtm

derekhiggins commented:
/approve
